src/schema/users.ts

```diff
@@ -1873,6 +1879,35 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         true,
       );
     },
+    plusGifterUser: async (_, __, ctx: AuthContext, info): Promise<GQLUser> => {
+      const { subscriptionFlags } = await ctx.con
+        .getRepository(User)
+        .findOneOrFail({
+          where: {
+            id: ctx.userId,
+          },
+          select: ['subscriptionFlags'],
+        });
+
+      if (!subscriptionFlags || !isGiftedPlus(subscriptionFlags)) {
+        throw new ForbiddenError('User is not a gifted plus user');
+      }
+
+      return await graphorm.queryOneOrFail<GQLUser>(
+        ctx,
+        info,
+        (builder) => {
+          builder.queryBuilder = builder.queryBuilder
+            .andWhere(`${builder.alias}."id" = :userId`, {
+              userId: subscriptionFlags.gifterId,
+            })
+            .select(['id', 'username', 'name', 'image']);
```

Comment:
We don't have to limit the properties here, in this case, we will do it through the actual GraphQL query. Automatically, defined props are the only ones to get pulled.

Reply:
+1